### PR TITLE
Make WP Rig child theme friendly

### PR DESCRIPTION
**Pull request by ataylorme**
## Description

WP Rig throws fatal errors when it is used as a child theme

## List of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Checklist:
- [ ] This pull request relates to a ticket.
- [x] This code is tested.
- [ ] This change has been added to CHANGELOG.md
- [x] I want my code added to WP Rig.

### CODE REVIEW

**Comment by mor10** [comment]
Old man winter called to tell me to tell you to take a mandatory 24 hrs off to relax and enjoy life.

**Comment by ataylorme** [comment]
Says the guy checking WP Rig emails 😀

**Comment by bamadesigner** [comment]
I'm going to test if WP Rig is a parent...

**Comment by bamadesigner** [comment]
So. Yea. This change throws errors if WP Rig is a parent theme because its looking for the files in the child theme. :)

I'll play with this for a bit and find the best solution.

**Comment by ataylorme** [comment]
@bamadesigner I wonder if we can use [`is_child_theme()`](https://developer.wordpress.org/reference/functions/is_child_theme/) to determine if the current theme is a child theme and then use `get_template_directory` or `get_stylesheet_directory` as appropriate. Although I think it will retur `true` if any child theme is activate, which would not help with WP Rig is the parent theme.

File: inc/Styles/Component.php [hunk 1]
```diff
@@ -85,6 +85,18 @@ public function action_enqueue_styles() {
 			wp_enqueue_style( 'wp-rig-fonts', $google_fonts_url, array(), null ); // phpcs:ignore WordPress.WP.EnqueuedResourceParameters.MissingVersion
 		}
 
+		// Enqueue parent theme style if this is a child theme.
+		if ( is_child_theme() ) {
```
Comment (bamadesigner) [inline inc/Styles/Component.php]: 2 thoughts:

1) Isn't common practice to @import via the child theme stylesheet?
2) But more importantly: we don't have any CSS in our style.css anymore in v2.0?

File: functions.php [hunk 2]
```diff
@@ -14,13 +14,13 @@
 
 // Bail if requirements are not met.
 if ( version_compare( $GLOBALS['wp_version'], WP_RIG_MINIMUM_WP_VERSION, '<' ) || version_compare( phpversion(), WP_RIG_MINIMUM_PHP_VERSION, '<' ) ) {
-	require get_template_directory() . '/inc/back-compat.php';
+	require get_stylesheet_directory() . '/inc/back-compat.php';
```
Comment (bamadesigner) [inline functions.php]: If we want to make sure the WP Rig files are always called, we might have to replace all the usage of `get_template_directory()` with `dirname( __FILE__ )`.

**Comment by bamadesigner** [comment]
@ataylorme Yea. I tried `is_child_theme()` but it only returns true if a child theme is active.

We might have to use some straight up PHP functions to force WP Rig files to load no matter its hierarchy.

I pushed this commit which might take care of it: https://github.com/wprig/wprig/commit/3d227893d8a75f759c8e7331f48983f40edfc96e

**Comment by ataylorme** [comment]
@bamadesigner what if we put the code for both WP Rig as a child theme and WP Rig as a parent theme in `optional` and document how to use them

**Comment by felixarntz** [comment]
`optional` is used for optional templates. I don't think we should put other stuff there that people _might_ use.

Specifically, as discussed in a meeting before, I'd rather document that WP Rig is intended to be used as a parent theme, and if you want to build a child theme on it, that's on you.

**Comment by joseph-farruggio** [comment]
@mor10 - As a designer/developer making client sites, can you speak to how you see me using WP Rig across multiple projects?

- I'll want to version my projects in git
- I'll want to deploy my projects with git
- I'll want to pull in changes to the WP Rig repo to existing projects

I thought a WP Rig parent/child theme would solve that. I thought I would clone WP Rig to start a project and create a child theme to provide project specific changes. Having WP Rig's modern build process for each child theme is where the value is for me.

**Comment by mor10** [comment]
Closing. Feel free to reopen if you have opinions.

**Comment by bamadesigner** [comment]
@joseph-farruggio Hi! 

To sum up our thoughts: WP Rig works great as a parent theme but is not intended for use as a child theme. WP Rig is built to lay a solid theme foundation, which is great for a parent theme, but not for a child theme. A child theme is meant to add on or modify the foundation.



I thought a WP Rig parent/child theme would solve that. I thought I would clone WP Rig to start a project and create a child theme to provide project specific changes. Having WP Rig's modern build process for each child theme is where the value is for me.
Using WP Rig as a child theme would 

WP Rig is child theme friendly, and how you're envisioning its usage is a great method. WP Rig works great as a parent theme.

This PR title is a little confusing and the discussion was a little disjointed. Basically, the discussion revolved more around using WP Rig itself as the child theme with a different parent theme. Which is not really how its intended to be used, hence closing the PR.

**Comment by bamadesigner** [comment]
@joseph-farruggio Hi!

To sum up our thoughts: WP Rig works great as a parent theme but is not intended for use as a child theme. WP Rig is built to lay a solid theme foundation, which is great for a parent theme, but not for a child theme. A child theme is meant to only add on or modify the foundation.

It sounds like you have a great scenario/need for a parent/child theme relationship. But WP Rig would serve better as the parent/the foundation. And then you can use lean child themes to simply add modifications.

I hope this helps/makes sense. I'm around if you have any questions.

**Pull request closed without merging**